### ISSUE TITLE
Add oldest supported build dependencies to build matrix

### ISSUE THREAD

**Issue by lethosor** [DFHack/dfhack#2165]
We have occasionally inadvertently introduced build failures for some users by relying on features that were introduced in new versions of dependencies. These include:

* CMake
* Perl
* Sphinx

I think we could safely test old versions of all of these at once with a single additional entry in the GitHub Actions build matrix.

A followup idea: should we use the _newest_ version for other builds? We do for Sphinx already, but we rely on the system versions of most others.

**Comment by myk002** [comment]
I think these are good ideas. Better to find compatibility issues in CI than to cause pain for users. One build action with old deps and the others with most recent stable versions. As long as the build env is Linux, this should work fine.

There may not be a reason to add an additional build action. We could perhaps repurpose the gcc-4.8 build for the "old deps" version.

**Comment by lethosor** [comment]
Yeah, we could probably use the GCC 4.8 matrix entry. Good point. On Buildmaster, some dependencies in the GCC 4.8 image are older simply because the base image is older (it's based on Ubuntu 16.04, I think?), but we haven't _needed_ to use older dependencies for the corresponding GCC build on GitHub yet.
